MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/SnakingDirectionsRouteActivity.java

```diff
@@ -48,24 +48,28 @@
 import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.lineWidth;
 
 /**
- * Rather than showing the directions all at once, have it "snake" from the origin to destination
+ * Rather than showing the directions route all at once, have it "snake" from the origin to destination by showing the
+ * route one {@link LegStep} section at a time.
  */
-
 public class SnakingDirectionsRouteActivity extends AppCompatActivity
   implements OnMapReadyCallback {
 
   private static final float NAVIGATION_LINE_WIDTH = 6;
+  private static final float NAVIGATION_LINE_OPACITY = .8f;
   private static final String DRIVING_ROUTE_POLYLINE_LINE_LAYER_ID = "DRIVING_ROUTE_POLYLINE_LINE_LAYER_ID";
   private static final String DRIVING_ROUTE_POLYLINE_SOURCE_ID = "DRIVING_ROUTE_POLYLINE_SOURCE_ID";
-  private static final String TAG = "SnakingRouteActivity";
+  private static final int DRAW_SPEED_MILLISECONDS = 500;
   private MapView mapView;
-  private MapboxMap map;
-  private MapboxDirections client;
+  private MapboxMap mapboxMap;
+  private MapboxDirections mapboxDirectionsClient;
+  private Handler handler;
+  private Runnable runnable;
+
   // Origin point in Paris, France
-  private static final Point origin = Point.fromLngLat(2.35222, 48.856614);
+  private static final Point PARIS_ORIGIN_POINT = Point.fromLngLat(2.35222, 48.856614);
 
   // Destination point in Lyon, France
-  private static  Point destination = Point.fromLngLat(4.83565, 45.76404);
+  private static Point LYON_DESTINATION_POINT = Point.fromLngLat(4.83565, 45.76404);
```

Comment:
can both be made final and maybe group all static final fields together?

Reply:
done and done